docs/pages/configuration.rst

```diff
@@ -1,4 +1,204 @@
+
+Note that the wavemap ROS server prints warnings for all unrecognized params at startup. This can be helpful for debugging and to quickly find typos in config param names.
+
+To get started quickly, we recommend skimming through these :repo_file:`example configs <ros/wavemap_ros/config>`.
+For reference, an overview of all available config options is provided below.
+
+ROS server
+**********
+These settings control the general behavior of the ROS server.
+They are nested in the top level config under ``map/general``.
+
+.. doxygenstruct:: wavemap::WavemapServerConfig
+    :project: wavemap
+    :members:
+
+Map data structures
+*******************
+These settings control the data structure that is used to store the map.
+They are nested in the top level config under ``map/data_structure``.
+
+The following three data structures are fully supported by wavemap's ROS server: ``wavelet_octree``, ``hashed_wavelet_octree`` or ``hashed_chunked_wavelet_octree``. For general use, we recommend the ``hashed_wavelet_octree`` data structure. If you need the best possible performance, the ``hashed_chunked_wavelet_octree`` data structure is faster and uses less RAM. However, it is still under active development.
+
+Wavelet octree
+==============
+Selected by setting ``map/data_structure/type: "wavelet_octree"``.
+
+The ``wavelet_octree`` is the simplest of the wavelet-based data structures and stores the wavelet coefficients in a standard octree. This data structure can be useful in case you need the entire map to be contained in a single tree and don't mind sacrificing performance.
+
+.. doxygenstruct:: wavemap::WaveletOctreeConfig
+    :project: wavemap
+    :members:
+
+Hashed wavelet octree
+=====================
+Selected by setting ``map/data_structure/type: "hashed_wavelet_octree"``.
+
+The ``hashed_wavelet_octree`` combines the strengths of wavelet octrees with block-hashing. At the top level, the map is split into blocks which are accessed through a hash table. Each block in turn contains a small wavelet octree.
+
+.. doxygenstruct:: wavemap::HashedWaveletOctreeConfig
+    :project: wavemap
+    :members:
+
+Hashed chunked wavelet octree
+=============================
+Selected by setting ``map/data_structure/type: "hashed_chunked_wavelet_octree"``.
+
+The ``hashed_chunked_wavelet_octree`` is similar to the ``hashed_wavelet_octree``, but instead of storing all octree nodes separately it groups them into chunks.
+
+.. doxygenstruct:: wavemap::HashedChunkedWaveletOctreeConfig
+    :project: wavemap
+    :members:
+
+
+Measurement inputs
+******************
+These settings control the measurement input handlers.
+They are nested in the top level config under ``inputs``. Note that ``inputs`` takes a YAML list, and one input handler will be created for each item in the list.
```

Comment:
Is YAML list correct given all this is JSON?

Reply:
Assuming that people will primarily use wavemap with ROS, the settings would be specified in ROS config files (YAML).
The only place where we currently use JSON is when defining the schemas for wavemap configs, which are defined using the [jsonschema standard](https://json-schema.org/). These schemas are used for linting (pre-commit) and can be used to simplify config editing (in CLion), but don't affect wavemap's runtime.
Based on your comment I briefly considered changing "YAML list" to just "list", but I think it's good to keep the former s.t. users can quickly find the necessary syntax for YAML lists on Google. Let me know if you agree/disagree :)